File: solutions/go/luhn/1/luhn.go
package luhn

import (
    "strings"
    "unicode"
)

func Valid(id string) bool {
    id = strings.ReplaceAll(id," ","")
    if len(id)<=1{return false}
	runes := []rune(id)
    num := 3
    sum := 0
    for i := len(runes)-1;i>=0;i--{
        if (unicode.IsDigit(runes[i])){
            d := int(runes[i]-'0')
        	if num%2==0{
            	d = d*2
            	if d>9{d-=9}
        	}
            num++
        	sum += d
        }else{
            return false
        }
    }
    return sum%10 == 0
    
}

File: solutions/c/luhn/1/luhn.c
#include "luhn.h"

bool luhn(const char *num){
    int digits = 0;
    int s = 0;
    bool dd = false;
    int i = strlen(num)-1;
    while(i>=0){
        if (isspace(num[i])){
            i--;
            continue;
        }else if (!isdigit(num[i])){
            return false;
        }
        int d = num[i] - '0';
        if (dd){
            d *= 2;
            if (d>9) d-=9;
        }
        dd = !dd;
        s += d;
        i--;
        digits++;
    }
    return digits > 1 && s%10==0;
}